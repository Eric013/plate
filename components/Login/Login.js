import { Button, Form, Grid } from 'semantic-ui-react'

import Link from 'next/prefetch'

export default () => (
  <Grid padded centered>
    <Grid.Row>
      <Grid.Column mobile={16} tablet={10} computer={6} largeScreen={6}>
        <h3>Login</h3>
        <Form onSubmit={(e) => e.preventDefault()}>
          <Form.Input type='text' name='email' placeholder='email' label='Email' />
          <Form.Input type='password' name='password' placeholder='password' label='Password' />
          <Button primary fluid>Login</Button>
        </Form>
      </Grid.Column>
    </Grid.Row>
    <Grid.Row>
      <Grid.Column mobile={16} tablet={10} computer={6} largeScreen={6}>
        <Link href='/forgotPassword'><a>Forgot Password?</a></Link>
      </Grid.Column>
    </Grid.Row>
  </Grid>
)
