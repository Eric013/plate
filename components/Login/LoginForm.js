import { Field, reduxForm } from 'redux-form'
import { compose, graphql } from 'react-apollo'

import BorderedButton from '../BorderedButton/BorderedButton'
import Link from 'next/link'
import PropTypes from 'prop-types'
import React from 'react'
import RenderWhiteTextField from '../../utils/RenderWhiteTextField'
import cookie from 'react-cookie'
import { loginMutation } from '../../mutations/loginMutations'
import { loginValidations } from '../../validations/loginValidations'

const userLogin = async login => {
  const username = document.getElementById('username').value
  const token = await login(username)
  if (token.data.login === null) {
    alert('invalid login')
  } else {
    cookie.save('token', token.data.login, { path: '/' })
    window.location.href = '/'
  }
}

const LoginForm = ({ handleSubmit, login }) => {
  return (
    <div className="container-fluid">
      <div className="row full-height middle-xs middle-sm middle-md middle-lg center-xs center-sm center-md center-lg">
        <div className="col-xs-12 col-sm-12 col-md-6 col-lg-4">
          <h1 className="header-text">Login</h1>
          <form onSubmit={handleSubmit(() => userLogin(login))}>
            <Field
              name="username"
              id="username"
              component={RenderWhiteTextField}
              type="text"
              label="Username"
            />
            <Field
              name="password"
              id="password"
              component={RenderWhiteTextField}
              type="password"
              label="Password"
              style={{ marginBottom: 20 }}
            />
            <BorderedButton color="white" type="submit" label="Login" />
          </form>
          <div className="forgot-password">
            <Link prefetch href="/forgotpassword">
              <a>Forgot Password?</a>
            </Link>
          </div>
        </div>
      </div>
      <style jsx>
        {`
          .text-field-email {
            margin-bottom: 15px;
          }
          .text-field-password {
            margin-bottom: 15px;
          }
          .forgot-password {
            margin-top: 15px;
          }
          .forgot-password a {
            color: white;
            text-decoration: none;
          }
          .full-height {
            height: 90vh;
          }
          .header-text {
            color: white;
            margin-bottom: 50px;
          }
        `}
      </style>
    </div>
  )
}

LoginForm.propTypes = {
  handleSubmit: PropTypes.func,
  login: PropTypes.func
}

export default compose(
  reduxForm({
    form: 'loginForm',
    validate: loginValidations
  }),
  graphql(loginMutation, {
    props: ({ mutate }) => ({
      login: username => mutate({ variables: { username } })
    })
  })
)(LoginForm)
