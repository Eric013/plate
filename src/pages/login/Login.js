import * as firebase from 'firebase'

import { Button, Col, ControlLabel, FormControl, FormGroup, Grid, Panel, Row } from 'react-bootstrap'
import { Link, browserHistory } from 'react-router'
import React, { Component } from 'react'

import { StyleSheet } from 'aphrodite'
import { css } from 'aphrodite'
import { toastr } from 'react-redux-toastr'

export default class Login extends Component {
  userLogin(e) {
    e.preventDefault()

    const email = document.getElementById('email').value
    const password = document.getElementById('password').value

    const promise = firebase.auth().signInWithEmailAndPassword(email, password)

    promise
      .then(() => browserHistory.push('/'))
      .then(() =>
        toastr.success('Success', 'Welcome To Plate')
      )
      .catch((e) =>
        toastr.error('Error', e.message)
      )
  }

  render() {
    return (
      <Grid fluid>
        <Row>
          <Col xs={12} sm={8} smOffset={2} md={6} mdOffset={3} lg={4} lgOffset={4}>
            <Panel header={<h3>Login</h3>} bsStyle="primary" className={css(styles.loginMargin)}>
              <form onSubmit={this.userLogin}>
                <FormGroup>
                  <ControlLabel>Email:</ControlLabel>
                  <FormControl type="email" id="email" placeholder="Email" required />
                </FormGroup>
                <FormGroup>
                  <ControlLabel>Password:</ControlLabel>
                  <FormControl type="password" id="password" placeholder="Password" required />
                </FormGroup>
                <FormGroup>
                  <Button type="submit" bsStyle="primary" bsSize="large" block>Login</Button>
                </FormGroup>
              </form>
              <Link to="/forgotPassword">Forgot Password?</Link>
            </Panel>
          </Col>
        </Row>
      </Grid>
    )
  }
}

const styles = StyleSheet.create({
  loginMargin: {
    marginTop: '25%'
  }
})
