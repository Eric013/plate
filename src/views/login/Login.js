import React from 'react'
import { Grid, Cell } from 'radium-grid'
import { Card, CardTitle, CardText } from 'material-ui/Card'
import TextField from 'material-ui/TextField'
import RaisedButton from 'material-ui/RaisedButton';

const styles = {
  loginContainer: {
    justifyContent: "center"
  },

  padTop: {
    marginTop: 50,
    width: "95%"
  }
}

function Login() {
  return (
    <Grid smallCellWidth="1" mediumCellWidth="1/2" align="center" style={styles.loginContainer}>
      <Cell>
        <Card style={styles.padTop}>
          <CardTitle title="Login" subtitle="Welcome To Spendy" />
          <CardText>
            <TextField hintText="Username" fullWidth={true} />
            <br />
            <TextField hintText="Password" fullWidth={true} type="password" />
            <br />
            <RaisedButton label="Login" secondary={true} fullWidth={true} />
          </CardText>
        </Card>
      </Cell>
    </Grid>
  )
}

export default Login
