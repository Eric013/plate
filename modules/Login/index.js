import BackButton from "../../components/BackButton/BackButton";
import Header from "../../components/Header/Header";
import LoginForm from "./components/LoginForm";
import NavlessLayout from "../NavlessLayout/NavlessLayout";
import React from "react";

export default () => (
  <NavlessLayout>
    <Header title="Login" />
    <BackButton />
    <LoginForm />
  </NavlessLayout>
);
