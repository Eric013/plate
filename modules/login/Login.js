import { Card, Grid, Header, Message } from 'semantic-ui-react'

import { Component } from 'react'
import Link from 'next/prefetch'
import LoginForm from './components/LoginForm'
import Router from 'next/router'
import { observer } from 'mobx-react'
import store from './store/store'

const Login = observer(class Login extends Component {
  userLogin = (e) => {
    e.preventDefault()

    store.loading = true
    Router.push('/')
    store.loading = false
  }

  render() {
    return (
      <Grid centered padded>
        <Grid.Column computer={5} largeScreen={5} tablet={10} mobile={16} textAlign='center'>
          <Card fluid raised>
            <Card.Content>
              <Header as='h2' color='grey'>
                Login to your account
              </Header>
            </Card.Content>
            <Card.Content extra>
              <LoginForm
                userLogin={this.userLogin}
                errorState={store.errorState}
                errorMessage={store.errorMessage}
                loading={store.loading}
              />
              <Message>
                <Link href='/forgotPassword'>Forgot your password?</Link>
              </Message>
            </Card.Content>
          </Card>
        </Grid.Column>
      </Grid>
    )
  }
})

export default Login
