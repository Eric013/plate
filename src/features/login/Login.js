import * as firebase from 'firebase'

import { Card, Grid } from 'semantic-ui-react'
import { Link, browserHistory } from 'react-router'
import React, { Component } from 'react'

import LoginForm from './LoginForm'
import { observer } from 'mobx-react'
import store from './store/store'

const Login = observer(class Login extends Component {
  userLogin = (e) => {
    e.preventDefault()

    store.loading = true

    const email = document.getElementById('email').value
    const password = document.getElementById('password').value

    const promise = firebase.auth().signInWithEmailAndPassword(email, password)

    promise
      .then(() => store.loading = false)
      .then(() => browserHistory.push('/'))

      .catch((e) => {
        store.errorState = 'initial'
        store.errorMessage = e.message
        store.loading = false
      })
  }

  render() {
    return (
      <Grid centered padded>
        <Grid.Column computer={8} largeScreen={5} tablet={10} mobile={16}>
          <Card fluid raised>
            <Card.Content header='Login' />
            <Card.Content extra>
              <LoginForm
                userLogin={this.userLogin}
                errorState={store.errorState}
                errorMessage={store.errorMessage}
                loading={store.loading}
              />
              <br />
              <Link to='/forgotPassword'>Forgot your password?</Link>
            </Card.Content>
          </Card>
        </Grid.Column>
      </Grid>
    )
  }
})

export default Login
