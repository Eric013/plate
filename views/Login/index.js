import BackButton from '../../components/BackButton/BackButton'
import LoginForm from './components/LoginForm'
import React from 'react'

const Login = () => {
  return (
    <div>
      <BackButton color="white" />
      <div className="container">
        <LoginForm />
      </div>
    </div>
  )
}

export default Login
