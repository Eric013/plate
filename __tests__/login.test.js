/* global it, expect, describe */

import Login from '../pages/login'
import { render } from 'react-dom'
import renderer from 'react-test-renderer'

it('renders without crashing', () => {
  const div = document.createElement('div')
  render(<Login />, div)
})

describe('Login snapshot', () => {
  it('renders without crashing!"', () => {
    const component = renderer.create(<Login />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
