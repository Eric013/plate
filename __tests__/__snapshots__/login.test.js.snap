exports[`Login snapshot renders without crashing!" 1`] = `
<div>
  <div
    className="nav-bar"
    data-jsx={2548831492}>
    <div
      className="ui pointing top fixed menu">
      <a
        data-jsx={2548831492}
        href="/"
        onClick={[Function]}>
        <div
          className="link item"
          onClick={[Function]}>
          <img
            data-jsx={2548831492}
            src="/static/plateLogo.png" />
          <span
            className="logo-text"
            data-jsx={2548831492}>
            late
          </span>
        </div>
      </a>
      <div
        className="right menu">
        <a
          className="link item"
          onClick={[Function]}>
          <span
            data-jsx={2548831492}>
            MENU
          </span>
        </a>
      </div>
    </div>
    <div
      data-jsx={1706063930} />
  </div>
  <div
    className="login-form"
    data-jsx={1816327275}>
    <div
      className="ui centered grid">
      <div
        className="row">
        <div
          className="eight wide computer sixteen wide mobile sixteen wide tablet column">
          <h3
            data-jsx={1816327275}>
            Login Into Your Account
          </h3>
          <form
            className="ui form"
            onSubmit={[Function]}>
            <div
              className="field">
              <input
                data-jsx={1816327275}
                id="email"
                name="email"
                placeholder="email"
                required={true}
                type="text" />
            </div>
            <div
              className="field">
              <input
                data-jsx={1816327275}
                id="password"
                name="password"
                placeholder="password"
                required={true}
                type="password" />
            </div>
            <button
              className="ui primary button"
              onClick={[Function]}
              tabIndex={undefined}
              type="submit">
              Login
            </button>
            <p
              data-jsx={1816327275}>
              <a
                className="link"
                data-jsx={1816327275}
                href="#">
                Forgot Password?
              </a>
            </p>
          </form>
        </div>
        <div
          className="computer only one wide computer column">
          <div
            className="ui vertical divider">
            or
          </div>
        </div>
        <div
          className="seven wide computer sixteen wide mobile column">
          <div
            className="buttons"
            data-jsx={1816327275}>
            <button
              className="ui facebook button"
              onClick={[Function]}
              style={
                Object {
                  "marginBottom": "15px",
                  "width": "300px",
                }
              }
              tabIndex={undefined}>
              <i
                aria-hidden="true"
                className="facebook icon" />
               Sign in with Facebook
            </button>
            <br
              data-jsx={1816327275} />
            <button
              className="ui google plus button"
              onClick={[Function]}
              style={
                Object {
                  "marginBottom": "15px",
                  "width": "300px",
                }
              }
              tabIndex={undefined}>
              <i
                aria-hidden="true"
                className="google plus icon" />
               Sign in with Google Plus
            </button>
            <br
              data-jsx={1816327275} />
            <button
              className="ui twitter button"
              onClick={[Function]}
              style={
                Object {
                  "width": "300px",
                }
              }
              tabIndex={undefined}>
              <i
                aria-hidden="true"
                className="twitter icon" />
               Sign in with Twitter
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
